client/src/Auth/auth.api.ts

```diff
@@ -0,0 +1,77 @@
+import { AxiosRequestConfig } from 'axios';
+
+import Api from '../common/api';
+
+const loginURL = '/user/login';
+const registerURL = '/user/register';
+const serverInfoURL = '/info/';
+const userInfoURL = '/user/';
+
+interface Credentials {
+    username: string;
+    password: string;
+}
+
+interface RegisterResponse {
+    error: string;
+}
+
+export const onRegister = async (
+    data: Credentials,
+): Promise<RegisterResponse | undefined> => {
+    const requestConfig: AxiosRequestConfig = {
+        method: 'post',
+        url: registerURL,
+        data,
+    };
+    const response = await Api.request<RegisterResponse>(requestConfig);
+    if (response && response.data) {
+        return response.data;
+    }
+    return { error: response };
+};
+
+interface LoginResponse {
+    error?: string;
```

Comment:
According to this interface, response from api might include only one error field, without information about success data fields. The same issue concern RegisterResponse...
I suggest to define ApiErrorInterface and make union type with success response interface.